include/aws/http/connection.h

```diff
@@ -548,6 +544,26 @@ int aws_http2_connection_get_received_goaway(
     uint32_t *out_http2_error,
     uint32_t *out_last_stream_id);
 
+/**
+ * Increment the connection's flow-control window to keep data flowing (HTTP/2 only).
+ *
+ * If the connection was created with `manual_window_management` set true,
+ * the flow-control window of the connection will shrink as body data is received for all the streams created on it.
+ * (headers, padding, and other metadata do not affect the window).
+ * The initial connection flow-control window is 65,535.
+ * Once the connection's flow-control window reaches to 0, all the streams on the connection stop receiving any further
+ * data.
+ *
+ * If `manual_window_management` is false, this call will have no effect.
+ * The connection maintains its flow-control windows such that
+ * no back-pressure is applied and data arrives as fast as possible.
+ *
```

Comment:
Document that if the connection is closed this call has no effect

Document that this function can only fail due to invalid input

We might want to do a pass like this over similar functions that act on the living connection. With the thinking that: we don't want to raise errors sometimes because the connection is closing, if the user wouldn't do anything in response to that error

Reply:
Checked all the similar functions, I think we are good.
We fail the call when connection is closed, but they make sense in those case. Eg:
1. ping, user will wait for ping complete callback if we don't fail it hard
2. send goaway. In case user wants allow more streams, but connection closed for another goaway that set a Last-Stream-ID, there will be misunderstanding if we don't fail the call.

Reply:
Talked offline, convinced that user will not have different actions when the goaway is not sent as the connection already closed.